Update Glossary (#821)
freedomofpress/securedrop-docs: docs/appendices/glossary.rst

Glossary
========

A number of terms used in this guide, and in the `SecureDrop workflow
diagram <what_is_securedrop>`, are specific to SecureDrop. 
The list below attempts to enumerate and define these terms.

.. _glossary_administrator:

Administrator
-------------

> The Administrator installs, deploys, and maintains a SecureDrop instance. They connect to the Application and Monitor servers (using SSH) and to the Admin Interface (using Tor Browser) from an Admin Workstation. 

Reviewer: I wonder if specifying "(using SSH)" is unnecessary here.

At least during initial setup, and then potentially later if something breaks, they will connect directly via keyboard and monitor.

Author: Good point! I improved the description and also remember the firewall, which is also under the admins purview! See https://github.com/freedomofpress/securedrop-docs/pull/821/commits/955b9011ff081247026f7aa1bed405a31e715bfe

.. _glossary_admin_interface:

Admin Interface
---------------

The Admin Interface is the web-based interface Administrators use to add and manage SecureDrop users. It is hosted by the Application Server as an authenticated Onion Service. The Admin Interface is only accessed by an Administrator using Tor Browser on an Admin Workstation.

Please see the :doc:`relevant page</admin/reference/admin_interface>` for complete instructions on accessing and using the Admin Interface.

.. _glossary_admin_workstation:

Admin Workstation
-----------------

The Admin Workstation is a laptop running Qubes OS, initially configured as a SecureDrop Workstation but with additional tools and credentials added to allow an Administrator to install and maintain a SecureDrop instance, in particular the :doc:`securedrop-admin</admin/reference/securedrop_admin>` utility. An Admin Workstation is capable of connection to the web-based Admin Interface, and to the Application and Monitor servers using SSH.

.. _glossary_application_server:

Application Server
------------------

> The Application Server is a physical computer housed on-premises running the SecureDrop server application. Based on our :doc:`recommendations</admin/installation/hardware>`, this is typically a NUC or similar Mini PC. The Application hosts both

Reviewer: ```suggestion
The Application Server is a physical computer housed on-premises running the SecureDrop server application. Based on our :doc:`recommendations</admin/installation/hardware>`, this is typically a NUC or similar Mini PC. The Application hosts
```

Using "both" here doesn't really apply because the following list is more than two items.

.. _glossary_export_device:

Export Device
-------------

The Export Device is a physical USB flash drive used to transfer decrypted documents from a SecureDrop Workstation to a Journalist's everyday workstation, or to another computer for additional processing.

Please see the detailed security recommendations for the choice, configuration and use of your Export Device in the :doc:`journalist</journalist/journalist>` guide and in the :doc:`setup guide</admin/installation/provisioning_usb>`.

.. _glossary_journalist:

Journalist
----------

The Journalist uses SecureDrop to communicate with and download documents
submitted by the Source. Journalists do this by using SecureDrop
Inbox on a SecureDrop Workstation laptop to connect to the Application Server.

If a Journalist chooses to release any of these documents,
they can be prepared for publication on the SecureDrop Workstation before
being transferred to another computer.

Instructions for using SecureDrop as a Journalist are available in our
:doc:`Journalist Guide </journalist/journalist>`.

.. _glossary_journalist_alert_key:

> Journalist Alert Key

Reviewer: We refer to "Journalist Alert Key" here, but "Journalist Alert Public Key" further down. We should be consistent across the board.

Author: Previously the glossary had entries for "Journalist Alert Public Key" and "OSSEC Alert Public Key" which both referenced an associated but unnamed private key. The entry for "Submission Key" explicitly mentioned the Submission Public Key and Submission Private Key, both styled as sub-glossary terms. 

I've changed the former two entries to match the latter, where the main glossary entry is the _____ Key and then the ____ Public Key and Private Key are both mentioned in title case as terms that could be referred to elsewhere. I think it's important to do this for the alert keys because they are less sensitive but it is still a source of confusion that there is a public key and private key, that they are used for different things... without getting into a full PGP 101.  

Now that the email alerts page has better explanations of the creation of the keys and where they go (https://github.com/freedomofpress/securedrop-docs/pull/826), I added links to that page for both of the alert key glossary entries in https://github.com/freedomofpress/securedrop-docs/pull/821/commits/278a57863fe7a5895bc6cbf7657319096f8b62f3. Let me know if you think this helps/if you agree on the consistency since I've explained it. Another option would be change all three entries to "Key Pair" but I don't think that's necessary. Maybe being that explicit would be good though!

Author: In merging in the email alerts page, I also made a few edits there to reference the public/private alert keys explicitly as glossary terms where appropriate.

Reviewer: I think this change makes a lot of sense! I'm very happy with the main entry being "_____ Key" and referring to the _____ Public / Private keys in the descriptions (and throughout the docs) with Title Case. I agree that being explicit about when to use the public vs. private keys is very important, especially for particularly sensitive keys.

--------------------

The Journalist Alert Public Key is used for encrypting the daily alert email
that notifies Journalists about whether or not there has been
submission activity in the past 24 hours. The Journalist uses an associated
Journalist Alert Private Key to decrypt the alerts.

.. _glossary_landing_page:

Landing Page
------------

The Landing Page is the public-facing webpage for a SecureDrop instance. This
page is hosted as a standard (i.e. non-Tor) webpage on the news organization's

> website. It provides first instructions for potential Sources and includes

Reviewer: ```suggestion
website. It provides instructions for potential Sources and includes
```

the the onion address for the instance's :ref:`Source Interface <glossary_source_interface>`.

.. _glossary_monitor_server:

Monitor Server
--------------

> The Monitor Server is physical computer housed on-premises running the `OSSEC<https://www.ossec.net/>` intrusion detection software to monitor the Application Server and send out an email alert if something seems wrong. Like the Application server, it is typically a NUC or similar Mini PC. Only Administrators connect

Reviewer: ```suggestion
The Monitor Server is physical computer housed on-premises running the `OSSEC<https://www.ossec.net/>` intrusion detection software to monitor the Application Server and send out email alerts to notify Administrators about events, such as package installations or new login attempts. Like the Application server, it is typically a NUC or similar Mini PC. Only Administrators connect
```

Most of the alerts aren't sent when something is wrong, so we should better set expectations about what Admins will see

Author: I don't agree on this one. For glossary purposes, the raison d'etre for the Monitor Server is to host OSSEC to detect potential intrusions on the Application Server and alert administrators. 

We do not deploy OSSEC to alert admin about benign events. True, that is mostly what happens. But we have an admin reference separate page describing how to analyze and interpret OSSEC alerts and as long as OSSEC is being used, it's important for readers to understand the purpose of the Monitor Server. Setting expectations for admins for what to expect from OSSEC alerts in practice I think belongs elsewhere, maybe in introductory text for administrators, or post install maintenance (I think it's in there to some effect).

The phrase "something seems wrong" isn't great, but after naming it as "intrusion detection software" I thought it was OK to leave, but I think your suggestion does point to how this could be improved. See https://github.com/freedomofpress/securedrop-docs/pull/821/commits/5e5a25fa710992a716914a2d193ac6c73ccd2f48 I've made it more descriptive.

Reviewer: Well said! I think my distinction was more that it sends alerts about events that are largely informational in nature, which _could_ point to a potential intrusion or other serious issue, and that really it's up to the Admins to use make the ultimate judgement calls. The change you made in https://github.com/freedomofpress/securedrop-docs/commit/5e5a25fa710992a716914a2d193ac6c73ccd2f48 captures that beautifully.

to this server.

Instructions and tips for accessing the Monitor Server over SSH are on the :ref:`SSH access page<server SSH access>`.

.. _glossary_onion_service:

Onion Service
-------------

Tor Onion Services provide anonymous inbound connections to websites and other
servers exclusively over the Tor network. SecureDrop uses Onion
Services for the Admin Interface and Source Interface websites,
as well as for SSH access to the servers in SSH-over-Tor mode.

Onion Services can be only be accessed using Tor Browser. Rather than a traditional internet address, Onion Services are reached using a special onion address. For example,
``sdolvtfhatvsysc6l34d65ymdwxcujausv7k5jk4cy5ttzhjoi6fzvyd.onion`` is the onion
address for the Onion Service version of the SecureDrop website. 

Authenticated Onion Services require an additional authentication token be provided when connecting to the server. The Admin Interface, the API endpoint connected to by SecureDrop Inbox, and SSH access to the servers (in SSH-over-Tor mode) are all authenticated Onion Services.

Read more about `Onion Services in Tor's glossary
<https://support.torproject.org/onionservices/>`__.

.. _glossary_ossec_alert_key:

> OSSEC Alert Key

Reviewer: Same feedback re: Journalist Alert [Public] Key. We should make sure these terms are consistent

Author: See comment above re: Journalist key
